### PR TITLE
Controlplaneoperator: Don't set status.kubeconfig before it exists

### PR DESCRIPTION
**Pull request by alvaroaleman**
Currently, creating a cluster always results in a bunch of errors from
the hosted cluster reconciler because it got a 404 trying to fetch the
hosted controlplane kubeconfig. This happens because the hosted
controlplane operator sets status.kubeconfig before that secret actually
exists, whereas the hosted controlplane operator assumes that as soon as
that field is non-nil, it exists.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 1]
```diff
@@ -374,14 +375,21 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
...
-			Name: DefaultAdminKubeconfigName,
+			Name: kubeconfig.Name,
 			Key:  DefaultAdminKubeconfigKey,
 		}
+		if hostedControlPlane.Spec.KubeConfig != nil {
+			hostedControlPlane.Status.KubeConfig.Key = hostedControlPlane.Spec.KubeConfig.Key
+		}
 	}
+
 	hostedControlPlane.Status.Initialized = true
```
Comment (alvaroaleman) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:393]: Is it ok that this gets set to `true` even when .status.Kubeconfig might be nil?

Comment (csrwng) [reply control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:393]: This is part of the CAPI interface and my understanding is that it is used by the CAPI controller to decide whether to start watching nodes, so it really should be set to true only when the kube apiserver is available and a kubeconfig is available for CAPI to use.
https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/machine_controller.go#L231-L236
/cc @enxebre

**Pull request merged**